File: src/main/java/com/vladmihalcea/OneToMany/Post.scala
package com.vladmihalcea.OneToMany

import jakarta.persistence.{CascadeType, Entity, FetchType, GeneratedValue, GenerationType, Id, OneToMany, Table}
import java.{util => ju}

import org.eclipse.persistence.annotations.{IdValidation, PrimaryKey}

import scala.beans.BeanProperty

@Entity
@PrimaryKey(validation = IdValidation.NULL)
@Table(name = "post", schema = "OLINGO")
class Post(@BeanProperty val title: String) {
  @OneToMany(mappedBy = "post", cascade = Array(CascadeType.ALL), orphanRemoval = true)
  // Use Java collection types instead of Scala ones to make JPA happy
  final private val comments: ju.List[PostComment] = new ju.ArrayList[PostComment]

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private var id = 0L

  // Default constructor for hibernate
  // No public visibility required
  private def this() = this(null)

  //Constructors, getters and setters removed for brevity
  def addComment(comment: PostComment): Unit = {
    comments.add(comment)
    comment.setPost(this)
  }

  def removeComment(comment: PostComment): Unit = {
    comments.remove(comment)
    comment.setPost(null)
  }

  override def toString: String = s"Post: $id $id $title $comments."

}

File: src/main/java/com/vladmihalcea/OneToMany/Post.java
package com.vladmihalcea.OneToMany;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

import java.util.ArrayList;
import java.util.List;

@Entity(name = "Post")
@Table(name = "post", schema = "OLINGO")
public class Post {
    public Post() { }

    public Post(String title) {
        this.title = title;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @OneToMany(
            mappedBy = "post",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    private final List<PostComment> comments = new ArrayList<>();

    //Constructors, getters and setters removed for brevity

    public void addComment(PostComment comment) {
        comments.add(comment);
        comment.setPost(this);
    }

    public void removeComment(PostComment comment) {
        comments.remove(comment);
        comment.setPost(null);
    }

}